src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -119,5 +142,61 @@ private CodeAction TryCreateCodeAction(Document document, AddImportFixData fixDa
 
             throw ExceptionUtilities.Unreachable;
         }
+
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return new AddImportCodeFixAllProvider();
+        }
+
+        internal async Task<Document> FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, CancellationToken cancellationToken)
+        {
+            var getFixesTasks = diagnostics
+                .GroupBy(d => d.Location.SourceSpan)
+                .Select(g => GetFixesForDiagnosticsAsync(document, g.Key, g.AsImmutable(), true, false, cancellationToken));
+
+            var usableFixes = (await Task.WhenAll(getFixesTasks).ConfigureAwait(false))
+                .SelectMany(diagnosticFixes => diagnosticFixes)
+                // Diagnostics where there are more than one fix signify an ambiguity that we should leave for the user to sort out.
+                .Where(diagnosticFixes => diagnosticFixes.Fixes.Length == 1)
+                .Select(diagnosticFixes => diagnosticFixes.Fixes[0])
```

Comment:
I dont think we made it a contract that fix in fixes array is deterministic. just taking 0 blindly means, result of this fix might change as we undo and redo.